homeassistant/components/sensor/iota.py

```diff
@@ -0,0 +1,100 @@
+"""
+Support for IOTA wallets.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/iota
+"""
+import logging
+
+from homeassistant.components.iota import DOMAIN as IOTA_DOMAIN, IotaDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['iota']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the IOTA sensor."""
+    # Add sensors for wallet balance
+    iota_config = hass.data[IOTA_DOMAIN]
+    sensors = [IotaBalanceSensor(wallet, iota_config)
+               for wallet in iota_config['wallets']]
+
+    # Add sensor for node information
+    sensors.append(IotaNodeSensor(iota_config=iota_config))
+
+    add_devices(sensors)
+
+
+class IotaBalanceSensor(IotaDevice):
+    """Implement an IOTA sensor for displaying wallets balance."""
+
+    def __init__(self, wallet_config, iota_config):
+        """Initialize the sensor."""
+        super().__init__(name=wallet_config['name'],
+                         seed=wallet_config['seed'],
+                         iri=iota_config['iri'],
+                         is_testnet=iota_config['is_testnet'])
+        self._state = 0
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} Balance'.format(self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return 'IOTA'
+
+    def update(self):
+        """Fetch new balance from IRI."""
+        self._state = self.api.get_inputs()['totalBalance']
+
+
+class IotaNodeSensor(IotaDevice):
```

Comment:
I assume that this should be a binary sensor. A node can be offline or online.

Reply:
Nope, a node can have multiple attributes, which are changing over time (https://iota.readme.io/v1.2.0/reference#getnodeinfo)

Reply:
Attributes and state are not the same. A binary sensor can have attributes as well. If a node is always present then it's doesn't make sense.

There are a dozen values available from getNodeInfo. Those details should be exposed as attributes.

Reply:
They are:
https://github.com/jinnerbichler/home-assistant/blob/iota-wallet/homeassistant/components/sensor/iota.py#L94

Reply:
Information about the node is part of the attributes of the node sensor and should be part of the wallet.